tst/testinstall/function.tst

```diff
@@ -339,5 +339,29 @@ function ( x ) return ([ [ x ] ]{[ 1 ]}{[ 1 ]})[1, 1]; end
 gap> funcloop(x -> ([ [ x ] ]{[ 1 ]}{[ 1 ]}){[ 1 ]}); # EXPR_ELMS_LIST
 function ( x ) return ([ [ x ] ]{[ 1 ]}{[ 1 ]}){[ 1 ]}; end
 
+# Test functions with very large lists
+gap> funcstr := String(List([1..2097151], x -> x));;
+gap> funcstr := Concatenation("func := function() return ", funcstr, "; end;");;
+gap> Read(InputTextString(funcstr));;
+gap> func() = [1..2097151];
+true
+gap> funcstr := String(List([1..2097152], x -> x));;
+gap> funcstr := Concatenation("func := function() return ", funcstr, "; end;");;
+gap> Read(InputTextString(funcstr));;
+Error, function too large for parser
+
+# Test functions with very large records
+gap> r := rec();; for x in [1..2097150/2] do r.(x) := x; od;;
```

Comment:
Running the above code changes the behavior of GAP in a way that breaks another test later in the test suite. In a nutshell:
```
gap> r:=rec();
rec(  )
gap> \.(r, 1000000);
Error, Record Element: <rnam> must be a valid rnam (not the integer 1000000)
not in any function at *stdin*:2
type 'quit;' to quit to outer loop
brk>
gap> r2 := rec();; for x in [1..2097150/2] do r2.(x) := x; od;;
gap> \.(r, 1000000);
Error, Record Element: '<rec>.992517' must have an assigned value
not in any function at *stdin*:5
type 'quit;' to quit to outer loop
```


Possible fix: modify `tst/testinstall/recordname.tst` to use, say `MAX_SIZE_LIST_INTERNAL` instead of `1000000`, should fix this...

Reply:
I've adjusted this test now, let's see if that is enough.